doc/developer/design/20220404_metadata_storage.md

```diff
@@ -0,0 +1,304 @@
+# Metadata database for Materialize Platform
+
+## Summary
+
+Each [Materialize Platform layer](../platform/architecture-db.md) in each
+customer environment in Materialize Platform needs to durably record its
+metadata. What database to use is an open question. The correctness,
+performance, scalability, reliability, and availability goals of Materialize
+Platform heavily constrain our options. This document details those constraints
+and proposes a candidate system for further evaluation.
+
+## Constraints
+
+### Hard requirements
+
+The following are the non-negotiable requirements that flow from Materialize
+Platform's product goals.
+
+* **Resilience to single availability zone outages.** Materialize Platform must
```

Comment:
One take, btw, is that ... we punt on handling AZ failures at all for now, and just roll with the SQLite on EBS solution that we currently have. No possibility of tolerating AZ failures, but also literally no additional development work required from the cloud team. (Many operational concerns from @alex-hunt-materialize, of course, but the code to allocate PVCs is already written.)

Reply:
> we punt on handling AZ failures at all for now

If this is a possibility, I think we should definitely consider it on the table. 

> and just roll with the SQLite on EBS solution that we currently have

This I'm less sanguine about, due to the same operational concerns that @alex-hunt-materialize has already raised -- but I think that work required to shift from SQLite-on-EBS to something else that can go down with an AZ is _much_ less extensive.

Reply:
> If this is a possibility, I think we should definitely consider it on the table.

I think we need @JLDLaughlin and @heeringa to weigh in on whether "tolerating single node AZ failures" is a hard requirement for v1 of Materialize Platform.

Reply:
> we punt on handling AZ failures at all for now 

Would this inherently put Jepsen testing outcomes at risk?

> but I think that work required to shift from SQLite-on-EBS to something else that can go down with an AZ is much less extensive.

What would be a rough timeline for a project like this? (Also thinking of Jepsen testing here, which I believe we want to engage in late summer/early fall.)

Reply:
> Would this inherently put Jepsen testing outcomes at risk?

My understanding is that it would not, because Jepsen doesn't simulate an AZ going down, but it would put our longer-term uptime claims at risk. AWS only puts a 99.5% uptime SLA on single instances and I think that's probably comparable to what you could expect from an AZ (but take that with a grain of salt, I don't know for certain).

> What would be a rough timeline for a project like this? (Also thinking of Jepsen testing here, which I believe we want to engage in late summer/early fall.)

I would surmise that the work already in process around `Stash` would let us trivially switch to RDS or Aurora, which provide solutions to ☝🏽 , but @benesch would have to confirm?

Reply:
> My understanding is that it would not, because Jepsen doesn't simulate an AZ going down, but it would put our longer-term uptime claims at risk. AWS only puts a 99.5% uptime SLA on single instances and I think that's probably comparable to what you could expect from an AZ (but take that with a grain of salt, I don't know for certain).

That's right. Jepsen largely evaluates you against the claims you make. If you don't claim to be tolerant to an AZ failure, then you won't be tested against it—though it might be called out qualitatively as a limitation.